api/handler/v1/runtime.go

```diff
@@ -971,6 +980,87 @@ func (sv *RuntimeServiceServer) BackupDryRun(ctx context.Context, req *pb.Backup
 	}, nil
 }
 
+func (sv *RuntimeServiceServer) Backup(ctx context.Context, req *pb.BackupRequest) (*pb.BackupResponse, error) {
+	projectSpec, err := sv.getProjectSpec(req.ProjectName)
+	if err != nil {
+		return nil, err
+	}
+
+	namespaceRepo := sv.namespaceRepoFactory.New(projectSpec)
+	namespaceSpec, err := namespaceRepo.GetByName(req.Namespace)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "%s: namespace %s not found", err.Error(), req.Namespace)
+	}
+
+	resourceSpec, err := sv.resourceSvc.ReadResource(ctx, namespaceSpec, req.DatastoreName, req.ResourceName)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "%s: failed to read resource %s", err.Error(), req.ResourceName)
+	}
+
+	var jobSpecs []models.JobSpec
+	jobSpec, err := sv.jobSvc.GetByDestination(projectSpec, resourceSpec.URN)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "error while getting job: %v", err)
+	}
+	jobSpecs = append(jobSpecs, jobSpec)
+
+	if !req.IgnoreDownstream {
+		downstreamSpecs, err := sv.jobSvc.GetDownstream(ctx, projectSpec, jobSpec.Name)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "error while getting job downstream: %v", err)
+		}
+		jobSpecs = append(jobSpecs, downstreamSpecs...)
+	}
+
+	backupRequest := models.BackupRequest{
+		ResourceName:     req.ResourceName,
+		Project:          projectSpec,
+		Namespace:        namespaceSpec,
+		Description:      req.Description,
+		IgnoreDownstream: req.IgnoreDownstream,
+		DryRun:           false,
+		Config:           req.Config,
+		BackupTime:       time.Now(),
```

Comment:
actually BackupTime is only being used in `/ext/datastore/bigquery/table`, when updating the backup expiry. i can move this to just before the copy table started in bigquery package.

Reply:
I have moved the backup init to `/datastore/service.go`, need your help to verify if it is proper. I didn't move it under bigquery package as it will be tricky to be tested.